Statsdash/report.py

```diff
@@ -0,0 +1,494 @@
+            summary_table=summary_table,
+            geo_table=country_table,
+            site_summary=site_table,
+            top_articles=article_table,
+            traffic_table=traffic_table,	
+            referrals=referring_site_table,
+            device_table=device_table,
+            social_table=social_table, 		
+        )
+        return html		
+        
+        
+    def send_email(self, html):
+        """
+        Send html email using config parameters
+        """
+        html = transform(html) #inline css using premailer
+        msg = MIMEMultipart('alternative')
+        msg.set_charset('utf8')
+        msg['Subject'] = self.get_subject()
+        msg['From'] = config.SEND_FROM
+        msg['To'] = ', '.join(self.recipients)
+        text_part = MIMEText("Please open with an HTML-enabled Email client.", 'plain')
+        html_part = MIMEText(html.encode("utf-8"), 'html')
+
+
+        if self.imgdata:
+            img_part = MIMEImage(self.imgdata, 'png')
+            img_part.add_header('Content-ID', '<graph>')
+            msg.attach(img_part)
+                    
+        msg.attach(text_part)
+        msg.attach(html_part)
+        
+        sender = smtplib.SMTP(config.SMTP_ADDRESS)
+        sender.sendmail(config.SEND_FROM, self.recipients, msg.as_string())
+    
+        sender.quit()
+        
+        
+        
+			
+class AnalyticsSocialReport(Report):
+    
+    def __init__(self, sites, period, recipients, frequency, subject):
+        super(AnalyticsSocialReport, self).__init__(sites, period, recipients, frequency, subject)
+        self.sites = sites
+        self.period = period
+        self.recipients = recipients
+        self.frequency = frequency
+        self.subject = subject		
+        self.data = AnalyticsData(self.sites, self.period, self.frequency)    
+        self.warning_sites = []
+        self.template = self.env.get_template("GA/social_report.html")
+        self.imgdata = None
+        
+        logger.debug("Running analytics social report")
+		
+    def check_data_availability(self, override=False):
+        """
+        incomplete
+        - could this be in main report
+        - does it need override or in scheduler should it only be called if override is false? 
+        - is there a better way than a function to call a function??
+        """
+        check = self.data.check_available_data()
+        if check["result"]:
+            return True
+        else:
+            logger.debug("Data not available for: %s" % check["site"])
+            if override:
+                self.warning_sites = check["site"]
+                return True
+            else:
+                return False	
+                
+    def get_site(self):
+        if len(self.sites) == 1:
+            return self.sites[0]
+        #TO DO: make this a check against total number of sites 
+        elif len(self.sites) == len(ga_config.TABLES.keys()):
+            return "Gamer Network"
```

Comment:
This is pretty Gamer Network specific!  I see the need for using something as an 'all sites' flag, but could it be something generic like `"ALL_SITES"` - this would need to change in a fair few places from what I can see.